Feat/consul smtp

frisonisland opened the pull request:
### Goal

Expose SMTP service through Consul as carbonio-mta service.

* Add systemd sidecar service definition (carbonio-mta-sidecar.service)
* Add service definition (carbonio-mta.hcl)
* Script to bootstrap service and define policies (carbonio-mta)

frisonisland committed:
chore: expose smtp through consul

frisonisland committed:
chore: remove access to mailbox for now
chore: remove PKGBUILD

frisonisland committed:
fix: policy name variable expansion in double quotes

frisonisland commented:
Hi @davide-baldo , I would like a feedback to check if requirements at policy level are met.
At the moment no intentions were added.

keshavbhatt commented on package/carbonio-mta:
```diff
@@ -0,0 +1,76 @@
+EXIT_CODE="$?"
+echo ""
+if [[ "${EXIT_CODE}" != "0" ]]; then
+  echo "cannot access to bootstrap token"
```
> "cannot access to bootstrap token"

Did you mean "cannot access the bootstrap token"?

frisonisland replied:
Yeah, the sentence is not syntactically correct as you pointed out.
The script instructions are the same as other (see mailbox), but will for sure update it if it improves readability!
In my opinion we should find a way to make these steps DRY across services where possible.

Thanks :)

frisonisland replied:
@keshavbhatt , now that I read it better and in the context, I think the meaning is that it is "not possible to access to the bootstrap-token service/command". That is why it's "to" and not "the", because the latter would mean access to "the token".

keshavbhatt replied:
Then it would make more sense to write "Unable to access bootstrap token" However, I will leave you with the decision. I personally will reconsider changing this elsewhere as well :)

keshavbhatt replied:
also consider "Unable to get bootstrap token" since this is exactly what we are doing in that subshell.

frisonisland replied:
Thanks @keshavbhatt , I am making some changes because there are already new guidelines to follow (see https://github.com/Zextras/consul-templates/) and I will stick to those descriptions

keshavbhatt commented on package/carbonio-mta:
```diff
@@ -0,0 +1,76 @@
+export -n SETUP_CONSUL_TOKEN
+
+POLICY_NAME="${SERVICE_NAME}-policy"
+POLICY_DESCRIPTION='Carbonio MTA service policy for config generation and for sidecar proxy'
```
> 'Carbonio MTA service policy for config generation and for sidecar proxy'

We can omit the later 'for' in this sentence to make it more readable.
Consider: 'Carbonio MTA service policy for config generation and sidecar proxy'

M0Rf30 replied:
All in all, except for some text or descrption enhancements, everything looks good to me

frisonisland replied:
Dropped the last for ;) Marking it as resolved

keshavbhatt reviewed (changes requested):
Made a few suggestions to make sentences more readable here and there.

frisonisland committed:
chore: policy description more readable

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Zextras/carbonio-mta?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Zextras/carbonio-mta?pullRequest=1) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Zextras/carbonio-mta?pullRequest=1) it.</sub>

davide-baldo commented:
@frisonisland 
You can double check against the official template, **it should be 90% the same** except the policies being in separated files and paths no longer using /etc/zextras but /etc/carbonio

https://github.com/Zextras/consul-templates

davide-baldo commented:
@frisonisland 
Each time the health check runs it creates few (3/4) lines, every 15s, it reduces MTA log readability, can you try to see if you can reduce the logging impact?
You could either change MTA configuration or use a dedicated script instead of a raw TCP connection.

frisonisland committed:
chore: policies.json and service-protocol.json files definitions
chore: update carbonio-mta to use policies file
chore: use new directory structure for mta consul files

frisonisland committed:
chore(carbonio-mta): add trap EXIT to avoid error when succesfull

frisonisland committed:
chore(carbonio-mta-sidecar.service): new line at end

frisonisland committed:
chore(carbonio-mta): remove unneeded override.conf

frisonisland commented:
> @frisonisland Each time the health check runs it creates few (3/4) lines, every 15s, it reduces MTA log readability, can you try to see if you can reduce the logging impact? You could either change MTA configuration or use a dedicated script instead of a raw TCP connection.

@davide-baldo  I took a look at the logs and it seems only /var/log/mail.log and /var/log/carbonio.log are affected, with many connection/disconnection usages due to health checks. Are we talking about these or also others?
A solution could be to drop TCP health check and instead check the postfix service is up, but no guarantee on port.

frisonisland committed:
chore(carbonio-mta.hcl): increase check every minute

frisonisland committed:
chore: add intentions allow for mailbox and files

keshavbhatt commented on package/carbonio-mta-sidecar.service:
```diff
@@ -0,0 +1,23 @@
+[Unit]
+Description=Carbonio mta sidecar proxy
+Documentation=https://docs.zextras.com/zextras-suite-documentation/latest/
```
Consider using this documentation URL like we do everywhere: 
https://docs.zextras.com/

If required, template repo should also be updated.

frisonisland committed:
chore(carbonio-mta-sidecar.service): update doc reference
